core/src/main/java/org/mule/runtime/core/processor/strategy/AbstractStreamPerEventProcessingStrategyFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.processor.strategy;
+
+import static reactor.core.publisher.Mono.just;
+
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.processor.Sink;
+import org.mule.runtime.core.api.processor.strategy.ProcessingStrategy;
+import org.mule.runtime.core.api.processor.strategy.ProcessingStrategyFactory;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.reactivestreams.Publisher;
+
+/**
+ * Abstract processing strategy that, rather than using a single shared stream, creates a stream per {@link Event}.
+ */
+abstract class AbstractStreamPerEventProcessingStrategyFactory extends AbstractProcessingStrategy {
+
+  @Override
+  public Sink createSink(FlowConstruct flowConstruct, Function<Publisher<Event>, Publisher<Event>> function) {
+    return new Sink() {
```

Comment:
other than this, don't we have some common sink implementations worth extracting to an abstract class?